python-checks/src/main/resources/org/sonar/l10n/py/rules/python/S6911.html

```diff
@@ -0,0 +1,78 @@
+<p>This rule raises an issue when a <code>tensorflow.function</code> depends on a global or free Python variable.</p>
+<h2>Why is this an issue?</h2>
+<p>When calling a <code>tensorflow.function</code> behind the scenes a <code>ConcreteFunction</code> is created everytime a new value is passed as
+argument. This is not the case with Python global variables, closure or nonlocal variables.</p>
+<p>This means the state and the result of the <code>tensorflow.function</code> may not be what is expected.</p>
+<pre>
+import tensorflow as tf
+
+@tf.function
+def addition():
+  return 1 + foo
+
+foo = 4
+addition() # tf.Tensor(5, shape=(), dtype=int32): on this first step we obtain the expected result
+
+foo = 10
+addition() # tf.Tensor(5, shape=(), dtype=int32): unexpected result of 5 instead of 11
+</pre>
+<p>As we can see in the example above the second time <code>addition</code> is called, we obtain the same result as the first call. This is due to the
+fact that between the 2 calls of <code>addition</code> the value of the argument passed to the function did not change. This result in the creation of
+a single <code>ConcreteFunction</code> during the first call of <code>addition</code>, with the value of foo set to 4.</p>
+<p>This is why it is a good practice to not use and mutate global variables or nonlocal variables inside of a <code>tensorflow.function</code>.</p>
+<h3>Exceptions</h3>
+<p>This rule will not raise an issue if the global or nonlocal variable is a <code>tensorflow.Variable</code>.</p>
```

Comment:
I believe this point of the RSPEC is not covered. We should make sure no issue is raised in such case (probably by checking if a `tf.Variable` is explicitly assigned to the variable). It's likely we'll still have to accept FPs though.